HarfangLab/harfanglab/ingest/parser.yml

```
@@ -142,6 +142,10 @@ stages:
           organization.id: "{{json_event.message.tenant}}"
           url.original: "{{json_event.message.details_url_request.url}}"
 
+      - set:
+          agent.id: "{{json_event.message.agents[0].agent_id}}"
+        filter: "{{json_event.message.agents | length > 0}}"
```

I think that a custom field is more suitable, as the list of agents is linked to a threat. So, it isn't an event from an agent, but it's an event about a detection with a list of agents.
```suggestion
      - set:
          harfanglab.agent_ids: "{{json_event.message.agents | map(attribute='agent_id') | list}"
        filter: "{{json_event.message.agents | length > 0}}"
```